lib/mongoid/extensions/regexp.rb

```diff
@@ -18,6 +18,7 @@ module ClassMethods
         #
         # @since 3.0.0
         def mongoize(object)
+          return nil if object.blank?
```

Comment:
This should be `nil?` rather than `blank?` (and this applies to the time extension as well), and can you please add a test that `''` mongoizes to `//` as it currently does?